theano/gpuarray/tests/test_dnn.py

```diff
@@ -2131,3 +2131,89 @@ def funcs(out, params):
                                            (i + 1) * len(cudnn_grads_layer)]
         for j, g in enumerate(cudnn_grads_layer):
             utt.assert_allclose(ref_grads_layer[j], g)
+
+
+def test_dnn_spatialtf_grid_generator():
+    if not dnn.dnn_available(test_ctx_name):
+        raise SkipTest(dnn.dnn_available.msg)
+    utt.seed_rng()
+
+    float_type = theano.config.floatX
+
+    from scipy import misc
+    f = misc.face(gray=True).astype(float_type)
+
+    # shape: (num_images, channels, height, width), equivalent to NCHW
+    nchannels = f.shape[2] if len(f.shape) == 3 else 1
+    assert (nchannels is not None)
+    grid_dims = (3, 3, 128, 128)
+
+    rotation = [[-1, 0, 0],
+                [0, -1, 0]]
+
+    theta = np.asarray(grid_dims[0] * [rotation], dtype=float_type)
+    theta_gpu = gpuarray_shared_constructor(theta)
+
+    def normalize_input(input):
+        # Scale input from [0, 255] to [0, 2]
+        scale_factor = 2 ** -7  # 1/128
+        input *= scale_factor
```

Comment:
Not important, but for better precision, I think you should directly apply division, instead of compute scale_factor and then apply a multiplication:
```python
input /= 128.0
```